### ISSUE TITLE
S3 didn't delete empty folder when delete all files in folder after upgrade to 3.66

### ISSUE THREAD

**Issue by xpader** [seaweedfs/seaweedfs#5565]
**Describe the bug**
In version 3.65 S3 API, when delete all file in a folder, empty folder will be delete by s3 list api. But after upgrade to 3.66, it's not work now, delete all files in folder, empty folder still there, call s3 list api, empty folder still there.

**System Setup**
- List the command line to start "weed master", "weed volume", "weed filer", "weed s3", "weed mount".
`weed server -master.volumeSizeLimitMB=1024 -volume.max=500 -ip=192.168.1.154 -ip.bind=0.0.0.0 -rack=ppdb2 -s3 -s3.allowEmptyFolder=false -s3.config=/etc/seaweedfs/s3.json`
- OS version
`Ubuntu 20.04, seaweedfs running in docker.`
- output of `weed version`
`version 30GB 3.66 3682eb929 linux amd64`
- if using filer, show the content of `filer.toml`
```
[filer.options]
# with http DELETE, by default the filer would check whether a folder is empty.
# recursive_delete will delete all sub folders and files, similar to "rm -Rf"
recursive_delete = false
#max_file_name_length = 255

####################################################
# The following are filer store options
####################################################

[leveldb2]
# local on disk, mostly for simple single-machine setup, fairly scalable
# faster than previous leveldb, recommended.
enabled = false
dir = "/data/filerldb2"


[mysql]  # or memsql, tidb
# CREATE TABLE IF NOT EXISTS `filemeta` (
#   `dirhash`   BIGINT NOT NULL       COMMENT 'first 64 bits of MD5 hash value of directory field',
#   `name`      VARCHAR(766) NOT NULL COMMENT 'directory or file name',
#   `directory` TEXT NOT NULL         COMMENT 'full path to parent directory',
#   `meta`      LONGBLOB,
#   PRIMARY KEY (`dirhash`, `name`)
# ) DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;

enabled = true
hostname = "192.168.1.154"
port = 3306
username = "seaweedfs"
password = "***"
database = "seaweedfs"              # create or use an existing database
connection_max_idle = 2
connection_max_open = 100
connection_max_lifetime_seconds = 0
interpolateParams = false
# if insert/upsert failing, you can disable upsert or update query syntax to match your RDBMS syntax:
enableUpsert = false
upsertQuery = """INSERT INTO `%s` (`dirhash`,`name`,`directory`,`meta`) VALUES (?,?,?,?) AS `new` ON DUPLICATE KEY UPDATE `meta` = `new`.`meta`"""
```
**Expected behavior**
Delete empty folder when call s3 list api, or like new doc said delete empty folder when delete all files in folder.
https://github.com/seaweedfs/seaweedfs/wiki/Amazon-S3-API#empty-folders

**Screenshots**
![WX20240507-115216](https://github.com/seaweedfs/seaweedfs/assets/8629617/6c46a374-0126-41a9-bab3-6915a784711b)

**Additional context**
I am using seaweedfs s3 api to store loki's log chunks, and retention enabled, loki using a folder for every unique tag combination, lots of expired logs are deleted everyday, and leaves a lot of empty directories.

**Comment by chrislusf** [comment]
fixed in 3.67

**Comment by xpader** [comment]
@chrislusf Thanks, for me, it finally has practical value.
